piperider_cli/configuration.py

```diff
@@ -285,6 +285,11 @@ def ask_for_datasource(self):
             answers = inquirer.prompt(questions, raise_keyboard_interrupt=True)
             return answers['datasource']
 
+    def delete_datasource(self, datasource):
+        if datasource in self.dataSources:
+            self.dataSources.remove(datasource)
+            self.dump(PIPERIDER_CONFIG_PATH)
```

Comment:
due to `dump` implementation, users' change of `profiler`, `table inclusion/exclusion`, and `tables` in `config.yml` would be overwritten by config template

Reply:
Implement a new method of `Configuration` class called `flush_datasource`. This method will only update the `dataSources` field in `.piperider/config.yml`. It won't overwrite the other part of `config.yml` Even the user comments in `config.yml` will be conserved.